loader/loader.go

```diff
@@ -68,6 +69,16 @@ type Options struct {
 	projectNameImperativelySet bool
 	// Profiles set profiles to enable
 	Profiles []string
+	// ResourceLoaders manages support for remote resources
+	ResourceLoaders []ResourceLoader
+}
+
+// ResourceLoader is a plugable remote resource resolver
+type ResourceLoader interface {
+	// Accept returns `true` is the resource reference matches ResourceLoader supported protocol(s)
+	Accept(path string) bool
+	// Load returns the path to a local copy of remote resource identified by `path`.
+	Load(ctx context.Context, path string) (string, error)
```

Comment:
Would it make sense to actually return the contents as `[]byte` here? Internally, loader implementations could do on-disk caching for performance, but externally it could be more of a "store"/VFS interface.

That'd let us use this consistently in all code-paths, aka have a `FilesystemLoader` as the default/fallback. (Not suggesting that as part of this PR, just looking forward.)

That'd make testing easier in some cases, and hopefully help address some of the implicit direct I/O that happens, e.g. `include` / `extends` / file references in secrets, `env_file`, etc

Reply:
I had this in mind first, but there are many places we also load sibling content, and as such we really need a path so we can search within parent folder.

Reply:
Right, I had the same thought and then ended up looking around and deciding that returning a local path might make more sense in the long run. Okay with both implementations though.

Reply:
Joining @milas on this, I would actually love to see some [Afero](https://github.com/spf13/afero) in action here. 
It looks like the [CopyOnWriteFs](https://github.com/spf13/afero#copyonwritefs) would look great for this use case.
And it would handle the case of sibling files too.

Reply:
Ah yeah the sibling content makes sense and needs more thought. I guess a good first step (independent of this) would be to virtualize/abstract the direct filesystem operations happening and could then offer an updated loader interface that works with that.

@ulyssessouza Good reminder about Afero, we could definitely make interfaces compatible with them in `compose-go` so it's easy to plug their implementations in. There's also the newer `io.FS` stuff in Go which might work since `compose-go` only reads but never writes.

Reply:
I agree it would be nice we could rely on some higher level abstraction like the `io.fs` package, but as this all results into a `types.ConfigDetails` which uses file paths as `string`s this won't be very relevant afaict (not to mention the required refactoring is far from being trivial)

Reply:
@ulyssessouza @milas we can't use such an abstraction as we need an actual file path to populate attributes we pass directly to engine API, like build context or bind mount source.